rllib/offline/dataset_reader.py

```diff
@@ -142,6 +191,22 @@ def next(self) -> SampleBatchType:
             # Columns like obs are compressed when written by DatasetWriter.
             d = from_json_data(d, self._ioctx.worker)
             count += d.count
-            ret.append(d)
+            ret.append(self._postprocess_if_needed(d))
         ret = concat_samples(ret)
         return ret
+
+    def _postprocess_if_needed(self, batch: SampleBatchType) -> SampleBatchType:
```

Comment:
this post processing was done in the JSON reader. Its probably done because there is no assumption that the data was produced by a similar policy as the one that the data was produced from. 

This post process function needs to be called somewhere -- we could call it inside of the sampler if needed, inside of env_runner. WDYT @gjoliver

Reply:
I see. man, this json read actually has access to the entire worker ...
ok, I'd add a TODO here and we can re-think about this.
I don't know how this works actually, like sometimes the offline dataset will already have advantages column ...